### PR TITLE
BCGOVSSIMO-7 Proof Details screen

### PR DESCRIPTION
**Pull request by nikita-zakharov-dsr**
# Summary of Changes

Add Proof Details History screen

# Related Issues

https://jira.dsr-corporation.com/browse/BCGOVSSIMO-7

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: core/App/screens/ProofDetailsHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      <View style={styles.content}>
+        <View style={{ marginVertical: 20 }}>
+          <Text style={{ color: 'white' }}>
+            <Text style={{ fontWeight: 'bold' }}>{connectionLabel}</Text> is sharing the following information from 2
+            credentials.
+          </Text>
+        </View>
+        <SharedProofData recordId={record.id} />
+      </View>
+      <View style={styles.footer}>
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetailsHistory.tsx:78]: There is no this button on the mockup - https://xd.adobe.com/view/993bddd5-51cd-435b-b1bf-6b2224de6d99-cc68/screen/a711deb6-83ec-4ff1-b8ce-aff3825b4f44/

File: core/App/screens/ProofDetailsHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    }
+  }, [navigation])
+
+  useEffect(() => {
+    if (!connection) return
+    navigation.setOptions({ title: connectionLabel }) //
+  }, [connection])
+
+  useEffect(() => {
+    console.log(record)
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetailsHistory.tsx:64]: why do we need it?

File: core/App/screens/ProofDetailsHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  }, [])
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.content}>
+        <View style={{ marginVertical: 20 }}>
+          <Text style={{ color: 'white' }}>
+            <Text style={{ fontWeight: 'bold' }}>{connectionLabel}</Text> is sharing the following information from 2
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetailsHistory.tsx:72]: Please define translations

**Review by Artemkaaas — CHANGES_REQUESTED** [review]

File: core/App/screens/ProofDetailsHistory.tsx (hunk 2 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetailsHistory.tsx]: fixed

File: core/App/screens/ProofDetailsHistory.tsx (hunk 3 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetailsHistory.tsx]: done

File: core/App/screens/ProofDetailsHistory.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import { ProofRequestsStackParams, Screens } from '../types/navigators'
+
+type ProofDetailsProps = StackScreenProps<ProofRequestsStackParams, Screens.ProofDetailsHistory>
+
+interface ProofDetailsHistoryProps {
+  record: ProofExchangeRecord
+  navigation: StackNavigationProp<ProofRequestsStackParams, Screens.ProofDetailsHistory>
+}
+
+const connectionLessLabel = 'Anonymous'
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetailsHistory.tsx:24]: it should be added to translations

File: core/App/screens/ProofDetails.tsx [hunk 5]
```diff
@@ -83,37 +105,55 @@ const VerifiedProof: React.FC<VerifiedProofProps> = ({ record, navigation, isHis
...
+        {/*  </View>*/}
+        {/*)}*/}
       </View>
-      <Collapsible collapsed={isCollapsed} collapsedHeight={collapsedHeight} enablePointerEvents={true}>
-        <View style={styles.content}>
-          <SharedProofData recordId={record.id} />
-        </View>
-      </Collapsible>
+      {/*</Collapsible>*/}
       {!isHistory && (
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetails.tsx:140]: I suppose `isHistory` falg is not needed anymore

File: core/App/screens/ProofDetails.tsx [hunk 6]
```diff
@@ -83,37 +105,55 @@ const VerifiedProof: React.FC<VerifiedProofProps> = ({ record, navigation, isHis
...
+        <View style={styles.header}>
+          <View style={styles.headerTitleContainer}>
+            <CheckInCircle {...{ height: 45, width: 45 }} />
+            <Text style={styles.headerTitle}>{t('Verifier.InformationReceived')}</Text>
+          </View>
+          <Text style={styles.headerDetails}>{t('Verifier.InformationReceivedDetails')}</Text>
         </View>
-        <Text style={styles.headerDetails}>{t('Verifier.InformationReceivedDetails')}</Text>
+      )}
+
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetails.tsx:124]: Can we create out custrom `Collapsible` componenton top and move related lines there?

File: core/App/screens/ProofDetailsHistory.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+  const connection = record.connectionId ? useConnectionById(record.connectionId) : undefined
+  const connectionLabel = useMemo(
+    () => (connection ? connection?.alias || connection?.theirLabel : connectionLessLabel),
+    [connection]
+  )
+
+  const [sharedProofDataItems, setSharedProofDataItems] = useState<GroupedSharedProofDataItem[]>([])
+
+  function onSharedProofDataLoad(data: GroupedSharedProofDataItem[]) {
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetailsHistory.tsx:55]: change to arrow function to match others

File: core/App/screens/ProofDetailsHistory.tsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    setSharedProofDataItems(data)
+  }
+
+  useEffect(() => {
+    navigation.setOptions({ title: connectionLabel }) //
+  }, [connectionLabel])
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.content}>
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetailsHistory.tsx:65]: Do we really need `container` and `content` views here? Can it be combined?

File: core/App/screens/ProofDetailsHistory.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        {sharedProofDataItems.length > 0 && (
+          <View style={styles.descriptionContainer}>
+            <Text style={styles.descriptionText}>
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetailsHistory.tsx:68]: Text block embeded into Text ? I suppose is should be View with row direction and two Texts inside

**Review by Artemkaaas — CHANGES_REQUESTED** [review]

File: core/App/screens/ProofDetailsHistory.tsx (hunk 4 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetailsHistory.tsx]: added

File: core/App/screens/ProofDetailsHistory.tsx (hunk 9 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetailsHistory.tsx:66]: Text supports nesting
https://reactnative.dev/docs/text

File: core/App/screens/ProofDetailsHistory.tsx (hunk 8 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetailsHistory.tsx]: fixed

File: core/App/screens/ProofDetailsHistory.tsx (hunk 7 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetailsHistory.tsx]: fixed

File: core/App/screens/ProofDetails.tsx (hunk 5 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetails.tsx]: fixed

File: core/App/screens/ProofDetails.tsx [hunk 10]
```diff
@@ -21,23 +22,39 @@ import { testIdWithKey } from '../utils/testable'
...
   navigation: StackNavigationProp<ProofRequestsStackParams, Screens.ProofDetails>
-  isHistory?: boolean
 }
 
 interface UnverifiedProofProps {
   record: ProofExchangeRecord
 }
 
-const VerifiedProof: React.FC<VerifiedProofProps> = ({ record, navigation, isHistory }: VerifiedProofProps) => {
+// const gradientSVG = `<svg xmlns="http://www.w3.org/2000/svg" width="600px" height="200px" >
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetails.tsx:37]: Should it me moved to `CollapsibleView` component?

File: core/App/screens/ProofDetails.tsx [hunk 11]
```diff
@@ -66,13 +83,17 @@ const VerifiedProof: React.FC<VerifiedProofProps> = ({ record, navigation, isHis
...
+      marginVertical: 20,
     },
     footerButton: {
       marginTop: 10,
     },
+    gradientContainer: { position: 'absolute', top: 0, width: '100%', height: '100%' },
   })
 
-  const [isCollapsed, setIsCollapsed] = useState<boolean>(!isHistory)
+  // const [isCollapsed, setIsCollapsed] = useState<boolean>(!isHistory)
```
Comment (Artemkaaas) [inline core/App/screens/ProofDetails.tsx:94]: Please remove all commented code

**Review by Artemkaaas — CHANGES_REQUESTED** [review]

File: core/App/screens/ProofDetails.tsx (hunk 10 above)
Comment (nikita-zakharov-dsr) [reply core/App/screens/ProofDetails.tsx]: done

**Review by Artemkaaas — APPROVED** [review]

**Pull request merged**